Identify first time contributors

OmarShehata opened the pull request:
This implements the feature described here https://github.com/AnalyticalGraphicsInc/cesium-concierge/issues/85.

I branched off of https://github.com/AnalyticalGraphicsInc/cesium-concierge/pull/118 because I needed the contributors URL fix from there. This just checks the `CONTRIBUTORS.md` in the repo/branch you're merging _into_. If the user's name isn't there, they are declared to be a first time external contributor and concierge will @ mention outreach team members. I didn't bother differentiating between internal/external contributors because if it's someone's first contribution they are 99% probably an external contributor. 

To do when merged:

* [ ] Updated templates in repo's to enable the firstContribution message.

cesium-concierge commented:
Thanks for the pull request @OmarShehata! 
* :heavy_check_mark: Signed CLA found. 
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium-concierge/blob/first-time/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

---

:sparkles: :sparkles: :sparkles:__I'm the Cesium Concierge and I'm here to test things.__ :sparkles: :sparkles: :sparkles:

ggetz commented on lib/templates/issueClosed.hbs:
```diff
@@ -1,3 +1,9 @@
+{{#if isFirstContribution}}
+Thanks for the awesome work @{{ userName }}! Your contribution is about to launch to millions of users with the next release :rocket:
```
```suggestion
Thanks for the awesome work @{{ userName }}! Your contribution is about to launch to millions of users with the next release. :rocket:
```

ggetz commented on README.md:
```diff
@@ -36,11 +36,12 @@ The table below describes all the possible configuration variables, as well as t
-| `listenPath` | `string` | Path on which to listen for incoming requests | X | `"/"`
+| `port` | `number` | Port on which to listen to incoming requests. | X | `5000`
+| `listenPath` | `string` | Path on which to listen for incoming requests. | X | `"/"`
+| `outreachUsers` | `string` | Username(s) to mention when someone makes their first contribution. Example:`@OmarShehata`. You can add multiple people or prepend it with a message this way. | X | _Disabled if not set._
```
` prepend it with a message this way` - a little vague, an example of this use case would help.

ggetz replied:
Actually, do we need this value in the config file at all? Why not just specify it in the first time contribution message templates?

ggetz commented on specs/lib/commentOnClosedIssueSpec.js:
```diff
@@ -190,12 +237,75 @@ describe('commentOnClosedIssue', function () {
             });
     });
+
+    it('commentOnClosedIssue._implementation posts about first timer\'s contribution.', function (done) {
```
Let's also test that the message don't get included when it's not their first contribution.

OmarShehata commented:
Thanks for the super speed review @ggetz ! And you were right, it's a lot simpler just listing outreach users in the template itself as opposed to adding a config. I added a test to make sure it doesn't post if a user is not a first time contributor, fixed that missing period in the template, and added a TODO bullet point in this PR to remember to update the templates in the relevant repositories.

ggetz commented:
This is good to go. Let's check https://github.com/AnalyticalGraphicsInc/cesium-concierge/pull/126 works before deploying this change as well. And, of course, update the relevant repos when you deploy as well!

cesium-concierge commented:
Thanks again for your contribution @OmarShehata! 

No one has commented on this pull request in 30 days. Maintainers, can you review, merge or close to keep things tidy?

 _I'm going to re-bump this in 30 days. If you'd like me to stop, just comment with `@cesium-concierge stop`. If you want me to start again, just delete the comment._

---

:sparkles: :sparkles: :sparkles:__I'm the Cesium Concierge and I'm here to test things.__ :sparkles: :sparkles: :sparkles:

OmarShehata commented:
Once deployed, this should start working right away, since the Cesium repo uses the default issueClosedTemplate which will tag me and Sarah. We can change that here or in the Cesium repo if we want to change who to tag.
